src/Configuration/Customers/data/hooks/useCustomerUsersTableData.js

```diff
@@ -56,8 +56,13 @@ const useCustomerUsersTableData = (enterpriseUuid) => {
         setIsLoading(false);
       }
     };
+
     if (enterpriseUuid) {
-      fetch();
+      if (args.filters.length && args.filters[0].value.length > 2) {
+        fetch();
+      } else if (!args.filters.length) {
+        fetch();
+      }
```

Comment:
nit: Can condense this logic
```suggestion
      if (!args.filters.length || args.filters[0].value.length < 3) {
        fetch();
      }
```